order/interfaces.go

```diff
@@ -792,14 +823,77 @@ func (b *Bid) ReservedValue(feeSchedule terms.FeeSchedule,
 	clearingPrice := FixedRatePremium(b.FixedRate)
 
 	return reservedValue(b, func(amt btcutil.Amount) btcutil.Amount {
+		premiumAmt := amt
+		if b.Details().AuctionType == BTCOutboundLiquidity {
+			premiumAmt += b.SelfChanBalance
+		}
+
 		delta, _, _ := takerDelta(
-			feeSchedule, clearingPrice, amt, b.SelfChanBalance,
-			b.LeaseDuration,
+			feeSchedule, clearingPrice, premiumAmt,
+			b.SelfChanBalance, b.LeaseDuration,
 		)
 		return delta
 	}, accountVersion)
 }
 
+// CheckOfferParams makes sure the offer parameters of a sidecar ticket are
+// valid and sane.
+func CheckOfferParams(auctionType AuctionType, capacity, pushAmt,
+	baseSupplyUnit btcutil.Amount) error {
+
+	if capacity == 0 || capacity%baseSupplyUnit != 0 {
+		return fmt.Errorf("channel capacity must be positive multiple "+
+			"of %d", baseSupplyUnit)
+	}
+
+	if auctionType == BTCInboundLiquidity && pushAmt > capacity {
+		return fmt.Errorf("self channel balance must be smaller than " +
+			"or equal to capacity")
+	}
+
+	if auctionType == BTCOutboundLiquidity {
+		// Only multiples of 100k sats are allowed in the outbound
+		// market.
+		if pushAmt == 0 || pushAmt%baseSupplyUnit != 0 {
+			return fmt.Errorf("self balance must be a positive "+
+				"multiple of %d", baseSupplyUnit)
+		}
+	}
+
+	return nil
+}
+
+// CheckOfferParamsForOrder makes sure that the order parameters in a
+// sidecar offer are formally valid, sane and match the order parameters.
+func CheckOfferParamsForOrder(auctionType AuctionType, offer sidecar.Offer,
```

Comment:
The doc for this function refers to sidecar offers specifically. But this PR moves this function from the sidecar package to the order package. Is the doc still correct?

Ditto for `CheckOfferParams`.